bin/all_sky_search/pycbc_fit_sngls_over_multiparam

```diff
@@ -15,8 +15,64 @@
 from __future__ import division
 
 import sys, h5py, argparse, logging, pycbc.version, numpy
+from scipy.stats import norm
 from pycbc.events import triggers
 
+def dist(i1, i2, parvals, smoothing_width):
+    dsq = 0
+    for v, s in zip(parvals, smoothing_width):
+        dsq += (v[i2] - v[i1]) ** 2.0 / s ** 2.0
+    return dsq ** 0.5
+
+def smooth_templates(nabove, invalphan, ntotal, template_idx,
+                      weights=None):
+    if weights is None: weights = numpy.ones_like(template_idx)
+    nabove_t_smoothed = numpy.average(nabove[template_idx], weights=weights)
+    ntotal_t_smoothed = numpy.average(ntotal[template_idx], weights=weights)
+    invalphan_mean = numpy.average(invalphan[template_idx], weights=weights)
+
+    # return nabove_smoothed, alpha_smoothed, ntotal_smoothed
+    return_tuple = (nabove_t_smoothed,
+                    nabove_t_smoothed / invalphan_mean,
+                    ntotal_t_smoothed)
+    return return_tuple
+
+def smooth_area(nabove, invalphan, ntotal, dists):
```

Comment:
Maybe the function name is a bit vague .. how about 'smooth_tophat' as this is a top hat window in some coordinates. 
Would be good to have 1 sentence or so of docstring explaining the format of the inputs - I think the first 3 are arrays covering the full set of templates and the last is an array - also over the full set of templates? - giving the distances of each to the (implicit) one template for which smoothed values are being made.

Reply:
Changed as suggested, docstring added